src/auth/models.hpp

```diff
@@ -44,15 +44,34 @@ enum class Permission : uint64_t {
 };
 // clang-format on
 
+// clang-format off
+enum class LabelPermission : uint64_t {
+  READ          = 1,
+  EDIT          = 1U << 1U,
+  CREATE_DELETE = 1U << 2U
+};
+// clang-format on
+
+constexpr inline uint64_t operator|(LabelPermission lhs, LabelPermission rhs) {
+  return static_cast<uint64_t>(lhs) | static_cast<uint64_t>(rhs);
+}
+
+constexpr inline uint64_t operator|(uint64_t lhs, LabelPermission rhs) { return lhs | static_cast<uint64_t>(rhs); }
+
+constexpr inline uint64_t operator&(uint64_t lhs, LabelPermission rhs) {
+  return (lhs & static_cast<uint64_t>(rhs)) != 0;
+}
+
+constexpr uint64_t LabelPermissionAll = memgraph::auth::LabelPermission::CREATE_DELETE |
+                                        memgraph::auth::LabelPermission::EDIT | memgraph::auth::LabelPermission::READ;
+constexpr uint64_t LabelPermissionMax = static_cast<uint64_t>(memgraph::auth::LabelPermission::CREATE_DELETE);
+constexpr uint64_t LabelPermissionMin = static_cast<uint64_t>(memgraph::auth::LabelPermission::READ);
+
 // Function that converts a permission to its string representation.
 std::string PermissionToString(Permission permission);
 
 // Class that indicates what permission level the user/role has.
-enum class PermissionLevel {
-  GRANT,
-  NEUTRAL,
-  DENY,
-};
+enum class PermissionLevel : short { GRANT, NEUTRAL, DENY };
```

Comment:
I would rather be specific on the type, maybe a suggestion suits you
```suggestion
enum class PermissionLevel : uint8_t { GRANT, NEUTRAL, DENY };
```